lib/redux/model/authentication/thunks.ts

```diff
@@ -63,14 +62,30 @@ export const login = createAppAsyncThunk(
                     return nullState;
                 }
 
+                let djName = "";
+                if (!backendData.dj_name || !backendData.real_name) {
+                    const { data: updateData, error: updateError } = await (updater(`djs/register`))({
+                        cognito_user_name: userResponse.Username,
+                        real_name: userResponse?.UserAttributes?.find(attr => attr.Name == 'name')?.Value,
+                        dj_name: userResponse?.UserAttributes?.find(attr => attr.Name == 'custom:dj-name')?.Value,
+                    });
+
+                    if (updateError) {
+                        toast.error("The backend is out of sync with the user database. This is fatal. Contact a site admin immediately.");
+                        return nullState;
+                    }
+
+                    djName = updateData.dj_name;
+                }
+
                 return {
                     authenticating: false,
                     isAuthenticated: true,
                     user: {
                         username: userResponse.Username || credentials.username,
-                        djName: userResponse.UserAttributes?.find(x => x.Name == "custom:dj-name")?.Value || "",
-                        djId: backendData.id,
-                        name: userResponse.UserAttributes?.find(x => x.Name == "name")?.Value || "",
+                        djName: backendData.dj_name || userResponse.UserAttributes?.find(x => x.Name == "custom:dj-name")?.Value || "",
+                        djId: Number(backendData.id),
+                        name: backendData.real_name || userResponse.UserAttributes?.find(x => x.Name == "name")?.Value || "",
```

Comment:
Could you extract these complex expressions into temporary named variables before passing them into the constructor?